### ISSUE TITLE
test_source_mtime_long_long fails on 32bit platforms

### ISSUE THREAD

**Issue by mimi1vx** [pytest-dev/pytest#5046]
fix in  #5045
- [x] Include a detailed description of the bug or suggestion
- [ ] `pip list` of the virtual environment you are using
- [x] pytest and operating system versions
- [x] Minimal example if possible

pytest-4.4.0 on openSUSE:Tumbleweed i586

```
[  348s] =================================== FAILURES ===================================
[  348s] _______________________ test_source_mtime_long_long[-1] ________________________
[  348s] 
[  348s] testdir = <Testdir local('/tmp/pytest-of-abuild/pytest-0/test_source_mtime_long_long0')>
[  348s] offset = -1
[  348s] 
[  348s]     @pytest.mark.parametrize("offset", [-1, +1])
[  348s]     def test_source_mtime_long_long(testdir, offset):
[  348s]         """Support modification dates after 2038 in rewritten files (#4903).
[  348s]     
[  348s]         pytest would crash with:
[  348s]     
[  348s]                 fp.write(struct.pack("<ll", mtime, size))
[  348s]             E   struct.error: argument out of range
[  348s]         """
[  348s]         p = testdir.makepyfile(
[  348s]             """
[  348s]             def test(): pass
[  348s]         """
[  348s]         )
[  348s]         # use unsigned long timestamp which overflows signed long,
[  348s]         # which was the cause of the bug
[  348s]         # +1 offset also tests masking of 0xFFFFFFFF
[  348s]         timestamp = 2 ** 32 + offset[  348s] =================================== FAILURES ===================================
[  348s] _______________________ test_source_mtime_long_long[-1] ________________________
[  348s] 
[  348s] testdir = <Testdir local('/tmp/pytest-of-abuild/pytest-0/test_source_mtime_long_long0')>
[  348s] offset = -1
[  348s] 
[  348s]     @pytest.mark.parametrize("offset", [-1, +1])
[  348s]     def test_source_mtime_long_long(testdir, offset):
[  348s]         """Support modification dates after 2038 in rewritten files (#4903).
[  348s]     
[  348s]         pytest would crash with:
[  348s]     
[  348s]                 fp.write(struct.pack("<ll", mtime, size))
[  348s]             E   struct.error: argument out of range
[  348s]         """
[  348s]         p = testdir.makepyfile(
[  348s]             """
[  348s]             def test(): pass
[  348s]         """
[  348s]         )
[  348s]         # use unsigned long timestamp which overflows signed long,
[  348s]         # which was the cause of the bug
[  348s]         # +1 offset also tests masking of 0xFFFFFFFF
[  348s]         timestamp = 2 ** 32 + offset
[  348s] >       os.utime(str(p), (timestamp, timestamp))
[  348s] E       OverflowError: Python int too large to convert to C long
[  348s] 
[  348s] /home/abuild/rpmbuild/BUILD/pytest-4.4.0/testing/test_assertrewrite.py:1253: OverflowError
[  348s] ________________________ test_source_mtime_long_long[1] ________________________
[  348s] 
[  348s] testdir = <Testdir local('/tmp/pytest-of-abuild/pytest-0/test_source_mtime_long_long1')>
[  348s] offset = 1
[  348s] 
[  348s]     @pytest.mark.parametrize("offset", [-1, +1])
[  348s]     def test_source_mtime_long_long(testdir, offset):
[  348s]         """Support modification dates after 2038 in rewritten files (#4903).
[  348s]     
[  348s]         pytest would crash with:
[  348s]     
[  348s]                 fp.write(struct.pack("<ll", mtime, size))
[  348s]             E   struct.error: argument out of range
[  348s]         """
[  348s]         p = testdir.makepyfile(
[  348s]             """
[  348s]             def test(): pass
[  348s]         """
[  348s]         )
[  348s]         # use unsigned long timestamp which overflows signed long,
[  348s]         # which was the cause of the bug
[  348s]         # +1 offset also tests masking of 0xFFFFFFFF
[  348s]         timestamp = 2 ** 32 + offset
[  348s] >       os.utime(str(p), (timestamp, timestamp))
[  348s] E       OverflowError: Python int too large to convert to C long
[  348s] 
[  348s] /home/abuild/rpmbuild/BUILD/pytest-4.4.0/testing/test_assertrewrite.py:1253: OverflowError
[  348s] >       os.utime(str(p), (timestamp, timestamp))
[  348s] E       OverflowError: Python int too large to convert to C long
[  348s] 
[  348s] /home/abuild/rpmbuild/BUILD/pytest-4.4.0/testing/test_assertrewrite.py:1253: OverflowError
[  348s] ________________________ test_source_mtime_long_long[1] ________________________
[  348s] 
[  348s] testdir = <Testdir local('/tmp/pytest-of-abuild/pytest-0/test_source_mtime_long_long1')>
[  348s] offset = 1
[  348s] 
[  348s]     @pytest.mark.parametrize("offset", [-1, +1])
[  348s]     def test_source_mtime_long_long(testdir, offset):
[  348s]         """Support modification dates after 2038 in rewritten files (#4903).
[  348s]     
[  348s]         pytest would crash with:
[  348s]     
[  348s]                 fp.write(struct.pack("<ll", mtime, size))
[  348s]             E   struct.error: argument out of range
[  348s]         """
[  348s]         p = testdir.makepyfile(
[  348s]             """
[  348s]             def test(): pass
[  348s]         """
[  348s]         )
[  348s]         # use unsigned long timestamp which overflows signed long,
[  348s]         # which was the cause of the bug
[  348s]         # +1 offset also tests masking of 0xFFFFFFFF
[  348s]         timestamp = 2 ** 32 + offset
[  348s] >       os.utime(str(p), (timestamp, timestamp))
[  348s] E       OverflowError: Python int too large to convert to C long
[  348s] 
[  348s] /home/abuild/rpmbuild/BUILD/pytest-4.4.0/testing/test_assertrewrite.py:1253: OverflowError
```

**Comment by Zac-HD** [comment]
What version of Python does this occur on?  If Python2, I suspect that this is simply not fixable for 32-bit operating systems (though we can add a `skipif` marker to the test).

**Comment by thmo** [comment]
Same issue with `platform linux2 -- Python 2.7.16, pytest-4.4.1, py-1.8.0, pluggy-0.9.0` on `i686`.

**Comment by Zac-HD** [comment]
Closed by #5045.
